### ISSUE TITLE
ImportError: No module named root

### ISSUE THREAD

**Issue by cdminigun** [google/grr#554]
>>> from grr_api_client import api
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/grr_api_client/api.py", line 7, in <module>
    from grr_api_client import root
  File "/usr/local/lib/python2.7/site-packages/grr_api_client/root.py", line 6, in <module>
    from grr.proto.api.root import user_management_pb2
ImportError: No module named root

Reproed on the newest version and on linux/OSX. 

Seems like it is tied to https://github.com/google/grr/commit/85a89ccdc14a66835b413249d127964baf940b04

**Comment by cdminigun** [comment]
Reverting to an older version of grr_api_client fixes the issue currently.

**Comment by mbushkov** [comment]
Can you please post the steps to reproduce? Which packages did you install and what did you try to run?

**Comment by cdminigun** [comment]
Python Version: (Though it occurred on Linux with python 2.7 as well) 
Python 2.7.14 (default, Sep 25 2017, 09:53:17)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)] on darwin

pip install grr-api-client
Version: grr-api-client-3.2.1.post0
Jump into the python interpreter and do:
from grr_api_client import api

**Comment by mbushkov** [comment]
Thanks for reporting! There's indeed an issue in grr-api-client's setup.py. Looking into this.

**Comment by cdminigun** [comment]
Seems like it applies to any 3.2 grr version. Also going back in version causes other issues.

**Comment by mbushkov** [comment]
Reopening until we publish new PIP packages.

**Comment by cdminigun** [comment]
Is there any thought for a stable vs release version of the API? We found that 3.2.0.post1 still works for now, but it'd be nice to have something stable.

**Comment by mbushkov** [comment]
Hi Christopher! Normally we release new versions of PIP packages after ensuring that all tests are green on Travis and that our end-to-end tests are passing. In this particular case, the error was not in the code itself but in the setup.py logic and so the bug slipped through. We'll review our release process to reduce the probability of such errors in the future.

As far as I know, PIP doesn't support having multiple branches of the same package. Consequently  there seems to be no easy way to implement stable/release branching with PIP packaging.

Unfortunately, no matter how many tests we have and how fine our release process is, the bugs may still slip into new versions of PIP packages. In GRR's setup.py files we always pin the dependencies to specific versions. One of possible workarounds for you may be to pin the grr-api-client dependency and then, when we release, repin it to a new version only after you make sure it works as expected in your environment. Would this kind of approach work for you?

**Comment by cdminigun** [comment]
I suppose I meant having previous pip versions backwards compatible, though I know that can be a tedious task. Thanks for fixing the bug.
